09_employment/compute_metrics_wage_ratio_2022_subgroup_new.do

```diff
@@ -0,0 +1,200 @@
+
+rename st state
+
+rename cnty county
+
+destring state, replace // make numeric for merging with MIT data
+
+cd `wages'
+
+/* merge living wage and QCEW data */
+merge m:1 state county using mit_living_wage-2022.dta
+
+*tab county state if _merge == 1*/// two AK counties missing from MIT data
+
+/* drop statewide obs */
+drop if _merge == 1
+
+/* only keep 1 adult, 2 children row */
+*keep if adults == "1 Adult" & children == "2 Children" // moved this earlier in the code to avoid a m:m merge in line 78
+
+/* drop duplicates (two counties repeated) */
+duplicates drop
+
+/* convert living hourly wage to weekly */
+gen weekly_living_wage = wage * 40
+
+/* get ratio (main metric) */
+gen average_to_living_wage_ratio = annualaverageweeklywage/weekly_living_wage
+
+/* create data quality flag 
+per discussion with Greg, >= 30 is 1, <30 is 3 */
+gen wage_ratio_quality = 1 if annualaverageestablishmentcount >= 30 & annualaverageestablishmentcount != .
+replace wage_ratio_quality = 3 if annualaverageestablishmentcount < 30 & annualaverageestablishmentcount != .
+replace wage_ratio_quality = . if annualaverageestablishmentcount == .
+
+/* test flag */
+tab wage_ratio_quality, missing
+
+/* put state and county in string with leading 0s */
+drop _merge
+
+** generate county name and state name
+*. countyfips, statefips(state) countyfips(county)
+
+gen new_state = string(state,"%02.0f")
+drop state
+rename new_state state
+
+gen new_county= string(county,"%03.0f")
+drop county
+rename new_county county
+
+/* check ratio */
+/*sum average_to_living_wage_ratio, det
+hist average_to_living_wage_ratio */
+/*tab county state if average_to_living_wage_ratio == 0 */
+
+gen new_ratio = string(average_to_living_wage_ratio)
+drop average_to_living_wage_ratio
+rename new_ratio average_to_living_wage_ratio
+
+/* replace 0 ratio with missing and replace data quality as missing */
+replace average_to_living_wage_ratio = "NA" if average_to_living_wage_ratio == "."
+replace wage_ratio_quality = 3 if average_to_living_wage_ratio == "NA"
+
+save "wage_ratio_final_2022_subgroup.dta",replace
+
+
+keep state county year average_to_living_wage_ratio wage_ratio_quality industry_type
+
+/* rename final variable here */
+
+*rename ratio_average_to_living_wage average_to_living_wage_ratio
+rename wage_ratio_quality ratio_living_wage_quality
+rename industry_type subgroup
+
+generate subgroup_type = 1 if subgroup == 1
+replace subgroup_type = 2 if subgroup != 1
+
+label define subg_type_lab 1 "all" 2 "industry"
+
+label values subgroup_type subg_type_lab
+
+/* export files */
+
+order year state county subgroup_type subgroup average_to_living_wage_ratio ratio_living_wage_quality
+
+export delimited using metrics_wage_ratio_2022_subgroup.csv, replace
+
+save "wage_ratio_final_2022_subgroup.dta",replace
+
+** add in county name and state name
+global crosswalk "C:\Users\KMartinchek\Documents\umf-2024\mobility-from-poverty\geographic-crosswalks\data"
```

Comment:
move up so directory paths are all in one place

Reply:
Moved.